blog(embertimes170): octane mental model (#833)
ember-learn/ember-blog: source/2020-12-18-the-ember-times-issue-170.md

---
title: The Ember Times - Issue No. 170
author: Ava Wroten, Chris Ng, the crowd
tags: Recent Posts, Newsletter, Performance, Ember Octane, Ember.js Times, Ember Times, 2020
alias : "blog/2020/12/18-the-ember-times-issue-170.html"
responsive: true
---

<SAYING-HELLO-IN-YOUR-FAVORITE-LANGUAGE> Emberistas! 🐹

Explore Ember/Glimmer component performance 🚅,

> Read the blog post on the Ember Octane mental model ✍️,

Reviewer: ```suggestion
<!--alex ignore nuts-->
Read the blog post on the Ember Octane mental model ✍️,
```

---

> ## [Blog: Ember Octane is a New Mental Model ✍️](https://v5.chriskrycho.com/journal/ember-octane-is-a-new-mental-model/)

Reviewer: ```suggestion
<!--alex ignore nuts-->
## [Blog: Ember Octane is a New Mental Model ✍️](https://v5.chriskrycho.com/journal/ember-octane-is-a-new-mental-model/)
```

> [Chris Krycho (@chriskrycho)](https://github.com/chriskrycho) wrote a [blog post](https://v5.chriskrycho.com/journal/ember-octane-is-a-new-mental-model/) on how Ember Octane is not a 1:1 translation from Ember Classic but rather a whole new way of thinking and designing your code. This new programming model means that some patterns that you’re used to using in the Classic paradigm simply do not work in the new paradigm.

Reviewer: ```suggestion
<!--alex ignore easy-->
[Chris Krycho (@chriskrycho)](https://github.com/chriskrycho) wrote a [blog post](https://v5.chriskrycho.com/journal/ember-octane-is-a-new-mental-model/) on how Ember Octane is not a 1:1 translation from Ember Classic but rather a whole new way of thinking and designing your code. This new programming model means that some patterns that you’re used to using in the Classic paradigm simply do not work in the new paradigm.
```

This didn't fail in CI but I think it does, cause I see the error in the alex extension on VSCode

> Chris goes through a couple examples of code that was fundamentally coupled to Ember Classic idioms such as the [Evented API](https://api.emberjs.com/ember/release/classes/Evented) and usage of `didReceiveAttrs`. The blog argues these cases where rare, it’s possible the abstraction would benefit from being reworked substantially towards the Octane idioms.

Reviewer: ```suggestion
Chris goes through a couple examples of code that was fundamentally coupled to Ember Classic idioms such as the [Evented API](https://api.emberjs.com/ember/release/classes/Evented) and usage of `didReceiveAttrs`. The blog argues that although these cases were rare, it’s possible the abstraction would benefit from being reworked substantially towards the Octane idioms.
```

Did you mean `were`? Wasn't sure.

Author: Yikes yes :D
